### PR TITLE
using poll instead of select in forwarder

### PR DESCRIPTION
**Pull request by jayendra13**
To resolve https://github.com/minrk/wurlitzer/issues/22

### CODE REVIEW

File: wurlitzer.py [hunk 1]
```diff
@@ -194,18 +198,21 @@ def forwarder():
...
-                    data = os.read(pipe, 1024)
+                    name = names[fd]
+                    data = os.read(fd, 1024)
                     if not data:
                         # pipe closed, stop polling it
-                        os.close(pipe)
-                        pipes.remove(pipe)
+                        os.close(fd)
+                        pipes.remove(fd)
+                        poller.unregister(fd)
```
Comment (minrk) [inline wurlitzer.py]: Maybe call unregister before close? Just in case unregister checks that the fd is valid.

Comment (jayendra13) [reply wurlitzer.py]: got it good catch, will change that

**Comment by jayendra13** [comment]
@minrk should there things needs to be debugged?

File: wurlitzer.py [hunk 2]
```diff
@@ -176,9 +176,13 @@ def forwarder():
             """Forward bytes on a pipe to stream messages"""
             draining = False
             flush_interval = 0
+            poller = select.poll()
+            for pipe_ in pipes: poller.register(pipe_)
```
Comment (minrk) [inline wurlitzer.py]: let's register these only for POLLIN so we only get readers, not writers:

```suggestion
            for pipe_ in pipes:
                poller.register(pipe_, select.POLLIN)
```

File: wurlitzer.py [hunk 3]
```diff
@@ -194,18 +198,21 @@ def forwarder():
...
                         continue
-                for pipe in r:
-                    if pipe == self._control_r:
+
+                for fd, flags in events:
+                    if fd == self._control_r:
                         draining = True
                         os.close(self._control_r)
                         pipes.remove(self._control_r)
+                        poller.unregister(self._control_r)
```
Comment (minrk) [inline wurlitzer.py]: unregister after close here, too

**Comment by jayendra13** [comment]
I am curious how all tests passed suddenly?  Any changes in unit-tests?

**Comment by minrk** [comment]
I restarted them and they were okay. seems to have been an intermittent issue on Travis

**Pull request merged**